com.unity.render-pipelines.core/Runtime/Volume/VolumeParameter.cs

```diff
@@ -1641,14 +1643,30 @@ internal override void Interp(VolumeParameter from, VolumeParameter to, float t)
     [Serializable]
     public class AnimationCurveParameter : VolumeParameter<AnimationCurve>
     {
+
         /// <summary>
         /// Creates a new <seealso cref="AnimationCurveParameter"/> instance.
         /// </summary>
         /// <param name="value">The initial value to be stored in the parameter</param>
         /// <param name="overrideState">The initial override state for the parameter</param>
         public AnimationCurveParameter(AnimationCurve value, bool overrideState = false)
-            : base(value, overrideState) { }
+            : base(value, overrideState)
+        {
+        }
 
-        // TODO: Curve interpolation
+        /// <summary>
+        /// Interpolates between two AnimationCurve values. Note that it will overwrite the values in lhsCurve,
+        /// whereas rhsCurve data will be unchanged. Thus, it is legal to call it as:
+        ///     stateParam.Interp(stateParam, toParam, interpFactor);
+        /// However, it should NOT be called when the lhsCurve parameter needs to be preserved.
```

Comment:
TBH, it's not ideal, and any solution is a bit messy. The optional workaround would be to change the HDRP code to detect the type of the parameter, and if it's a AnimationCurveParameter then call InterpAnimationCurve() directly instead of following the same overloaded Interp() path as everything else. That seemed messier than adding this rule to the animation curve. But if you would prefer to have it that way, I can make the change.

Reply:
Actually, thinking it through, that's probably a better way to do it. A lerp that overrides the data in the left side is fundamentally a different operation than blending lhs and rhs to return a new result. Better to make it explicit. Since it's an HDRP-specific exception, does it even make sense to have this functionality inside core?

Reply:
Fair point. But to me it seems that it can be usefull in any pipeline to actually interp curves. 

I dig into the code:
A VolumeComponent contains a readonly collection of VolumeParameter that is gathered at construction.
When the VolumeManager Update the volume stack, it start by resetting each one of this VolumeParameter to the one given by default values. Then it calls for each VolumeComponent its Override onto the current stack. This override call the Interp with first argument the one in the stack. 
So I looked at how it is done for a struct like. In FloatParameter, the underlying m_Value is changed. The current call being `stateParam.Interp(stateParam, toParam, interpFactor);` It means that we modify the first argument anyway (because it is also the one at left of the dot method accessor).
So regarding this, I bet that what you did is working in fact. But this comment is missleading. You should say something like "It should NOT be called when the lhsCurve parameter needs to be preserved. But current framework modify it anyway in VolumeComponent.Override for all type of VolumeParameters"

So if you just update this comment and also be sure that data in the scene are not modified, it will be enough in fact.

Reply:
Sounds good. In my last change, I actually went in and changed the stateParam.Interp() call to check for the type and explicitly call the curve interpolate function when needed, but I'll revert that change and update the comment as you describe.